File: CS/DXSample/ItemsControlEx.cs
using System;
using System.Linq;
using System.Windows;
using DevExpress.Xpf.Grid;
using System.Windows.Controls;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace DXSample
{
    public class ItemsControlEx : ItemsControl
    {
        public static readonly DependencyProperty AllItemsProperty =
            DependencyProperty.Register("AllItems", typeof(ObservableCollection<GridGroupSummaryData>), typeof(ItemsControlEx), new PropertyMetadata(null, OnAllItemsPropertyChanged));
        public static readonly DependencyProperty ColumnProperty =
            DependencyProperty.Register("Column", typeof(ColumnBase), typeof(ItemsControlEx), new PropertyMetadata(null, OnColumnPropertyChanged));
        public ObservableCollection<GridGroupSummaryData> AllItems
        {
            get { return (ObservableCollection<GridGroupSummaryData>)GetValue(AllItemsProperty); }
            set { SetValue(AllItemsProperty, value); }
        }
        public ColumnBase Column
        {
            get { return (ColumnBase)GetValue(ColumnProperty); }
            set { SetValue(ColumnProperty, value); }
        }

        static void OnAllItemsPropertyChanged(DependencyObject o, DependencyPropertyChangedEventArgs e)
        {
            ((ItemsControlEx)o).OnAllItemsChanged(e);
        }
        static void OnColumnPropertyChanged(DependencyObject o, DependencyPropertyChangedEventArgs e)
        {
            ((ItemsControlEx)o).OnColumnChanged(e);
        }
        void OnAllItemsChanged(DependencyPropertyChangedEventArgs e)
        {
            UnSubscribe(e.OldValue as ObservableCollection<GridGroupSummaryColumnData>);
            Subscribe(e.NewValue as ObservableCollection<GridGroupSummaryColumnData>);
            UpdateItemsSource();
        }
        void OnColumnChanged(DependencyPropertyChangedEventArgs e)
        {
            UpdateItemsSource();
        }
        void Subscribe(ObservableCollection<GridGroupSummaryColumnData> collection)
        {
            if (collection != null)
                collection.CollectionChanged += OnCollectionChanged;
        }
        void UnSubscribe(ObservableCollection<GridGroupSummaryColumnData> collection)
        {
            if (collection != null)
                collection.CollectionChanged -= OnCollectionChanged;
        }
        void OnCollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
        {
            UpdateItemsSource();
        }
        void UpdateItemsSource()
        {
            List<GridGroupSummaryData> data = new List<GridGroupSummaryData>();
            if (Column != null && AllItems != null)
            {
                foreach (var item in AllItems)
                    if (item.Column == Column)
                        data.Add(item);
            }
            ItemsSource = data;
        }
    }
}

File: VB/DXSample/ItemsControlEx.vb
Imports System
Imports System.Linq
Imports System.Windows
Imports DevExpress.Xpf.Grid
Imports System.Windows.Controls
Imports System.Collections.Generic
Imports System.Collections.ObjectModel

Namespace DXSample
	Public Class ItemsControlEx
		Inherits ItemsControl

		Public Shared ReadOnly AllItemsProperty As DependencyProperty = DependencyProperty.Register("AllItems", GetType(ObservableCollection(Of GridGroupSummaryData)), GetType(ItemsControlEx), New PropertyMetadata(Nothing, AddressOf OnAllItemsPropertyChanged))
		Public Shared ReadOnly ColumnProperty As DependencyProperty = DependencyProperty.Register("Column", GetType(ColumnBase), GetType(ItemsControlEx), New PropertyMetadata(Nothing, AddressOf OnColumnPropertyChanged))
		Public Property AllItems() As ObservableCollection(Of GridGroupSummaryData)
			Get
				Return CType(GetValue(AllItemsProperty), ObservableCollection(Of GridGroupSummaryData))
			End Get
			Set(ByVal value As ObservableCollection(Of GridGroupSummaryData))
				SetValue(AllItemsProperty, value)
			End Set
		End Property
		Public Property Column() As ColumnBase
			Get
				Return CType(GetValue(ColumnProperty), ColumnBase)
			End Get
			Set(ByVal value As ColumnBase)
				SetValue(ColumnProperty, value)
			End Set
		End Property

		Private Shared Sub OnAllItemsPropertyChanged(ByVal o As DependencyObject, ByVal e As DependencyPropertyChangedEventArgs)
			CType(o, ItemsControlEx).OnAllItemsChanged(e)
		End Sub
		Private Shared Sub OnColumnPropertyChanged(ByVal o As DependencyObject, ByVal e As DependencyPropertyChangedEventArgs)
			CType(o, ItemsControlEx).OnColumnChanged(e)
		End Sub
		Private Sub OnAllItemsChanged(ByVal e As DependencyPropertyChangedEventArgs)
			UnSubscribe(TryCast(e.OldValue, ObservableCollection(Of GridGroupSummaryColumnData)))
			Subscribe(TryCast(e.NewValue, ObservableCollection(Of GridGroupSummaryColumnData)))
			UpdateItemsSource()
		End Sub
		Private Sub OnColumnChanged(ByVal e As DependencyPropertyChangedEventArgs)
			UpdateItemsSource()
		End Sub
		Private Sub Subscribe(ByVal collection As ObservableCollection(Of GridGroupSummaryColumnData))
			If collection IsNot Nothing Then
				AddHandler collection.CollectionChanged, AddressOf OnCollectionChanged
			End If
		End Sub
		Private Sub UnSubscribe(ByVal collection As ObservableCollection(Of GridGroupSummaryColumnData))
			If collection IsNot Nothing Then
				RemoveHandler collection.CollectionChanged, AddressOf OnCollectionChanged
			End If
		End Sub
		Private Sub OnCollectionChanged(ByVal sender As Object, ByVal e As System.Collections.Specialized.NotifyCollectionChangedEventArgs)
			UpdateItemsSource()
		End Sub
		Private Sub UpdateItemsSource()
			Dim data As New List(Of GridGroupSummaryData)()
			If Column IsNot Nothing AndAlso AllItems IsNot Nothing Then
				For Each item In AllItems
					If item.Column Is Column Then
						data.Add(item)
					End If
				Next item
			End If
			ItemsSource = data
		End Sub
	End Class
End Namespace
